java/src/test/java/org/rocksdb/SequenceNumberAllocationTest.java

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+package org.rocksdb;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(Parameterized.class)
+public class SequenceNumberAllocationTest {
+  @Parameters(name = "{0}")
+  public static Iterable<Boolean> parameters() {
+    return Arrays.asList(
+      true,
+      false
+    );
+  }
+
+  @Parameter(0)
+  public Boolean disableWAL;
+
+  private Options options;
+  private WriteOptions writeOptions;
+  private FlushOptions flushOptions;
+
+  @ClassRule
+  public static final RocksNativeLibraryResource ROCKS_NATIVE_LIBRARY_RESOURCE = new RocksNativeLibraryResource();
+
+  @Rule
+  public TemporaryFolder dbFolder = new TemporaryFolder();
+
+  @Before
+  public void setup()
+  {
```

Comment:
formatting! The `{` should be on the line above etc...

Reply:
```
make format
$DEBUG_LEVEL is 1, $LIB_MODE is shared
Makefile:166: Warning: Compiling in debug mode. Don't use the resulting binary in production
build_tools/format-diff.sh
Checking format of uncommitted changes...
Nothing needs to be reformatted!
```

Reply:
Locally ran tests:
```
Running: org.rocksdb.SequenceNumberAllocationTest
	testSeqnoCloseWithoutFlush[true] OK
	testSeqnoDataWrittenAfterFlush[true] OK
	testSeqnoFlushAndClose[true] OK
	testSeqnoThreadInterruptAfterFlush[true] OK
	testSeqnoCloseWithoutFlush[false] OK
	testSeqnoDataWrittenAfterFlush[false] OK
	testSeqnoFlushAndClose[false] OK
	testSeqnoThreadInterruptAfterFlush[false] OK
Tests run: 8, Failures: 0, Errors: 0, Ignored: 0, Time elapsed: 2.617 sec

Time: 2.748

OK (8 tests)
```

Reply:
@adamretter is there anything else? if not can we please merge this, thanks!

Reply:
@adamretter rebased on main, would you please have another look? It would be great if we can wrap this up. Thank you! 😄